### DISCUSSION TITLE
Problems found when running Tasnet

### DISCUSSION THREAD

**Discussion by xs20211015** [asteroid-team/asteroid#659] [Q&A] [answered]
path：asteroid/egs/whamr/TasNet/local/prepare_data.sh
**Why is the code here commented：**
#!/bin/bash

wav_dir=tmp
out_dir=tmp
python_path=python

. utils/parse_options.sh

## Download WHAM noises
mkdir -p $out_dir
echo "Download WHAM noises into $out_dir"
# If downloading stalls for more than 20s, relaunch from previous state.
#wget -c --tries=0 --read-timeout=20 https://storage.googleapis.com/whisper-public/wham_noise.zip -P $out_dir
#
#
#echo "Download WHAMR scripts into $out_dir"
#wget https://storage.googleapis.com/whisper-public/whamr_scripts.tar.gz -P $out_dir
#tar -xzvf $out_dir/whamr_scripts.tar.gz -C $out_dir/
#mv $out_dir/whamr_scripts.tar.gz $out_dir/whamr_scripts
#
#wait
#
#echo "Unzip WHAM noises into $out_dir"
#mkdir -p logs
#unzip $out_dir/wham_noise.zip -d $out_dir >> logs/unzip_whamr.log

cd $out_dir/whamr_scripts
....

**Comment by mpariente** [comment]
I don't know, what doesn't work ?

**Reply by xs20211015** [reply]
Thank you very much for your reply.
When the code runs to this line: cd $out_dir/whamr_scripts, it shows an error that whamr_scripts cannot be found, and then I found that the previous data download steps have been commented out.

**Reply by mpariente** [reply]
Did you try uncommenting it ?

**Reply by xs20211015** [reply]
Yes, I uncommented the previous code. wham_noise.zip and whamr_scripts.tar.gz were downloaded and unpacked successfully.
But when running to the next step: $python_path -m pip install -r requirements.txt , a new error was encountered when installing dependencies:
`  Building wheel for pyroomacoustics (setup.py) ... done
  Created wheel for pyroomacoustics: filename=pyroomacoustics-0.3.1-cp39-cp39-linux_x86_64.whl size=621224 sha256=f8588aadda5d2bc5340d01cffd9cbc2c2c8e68e266a77616b3a6beb2eaa604b2
  Stored in directory: /root/.cache/pip/wheels/5b/e5/f6/47d4c4dfcb7e530f7805b548fda7ea12bbafaf8f5e95428782
  error: subprocess-exited-with-error
  
  × Building wheel for pandas (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> See above for output.
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  Building wheel for pandas (pyproject.toml) ... error
  ERROR: Failed building wheel for pandas
  Building wheel for scipy (pyproject.toml) ... done
  Created wheel for scipy: filename=scipy-1.5.2-cp39-cp39-linux_x86_64.whl size=61935559 sha256=c4054946918fb05c90524b878610c2461e245a91d392eebe61388fac45a29f7e
  Stored in directory: /root/.cache/pip/wheels/24/76/36/4147d04693ab616d9e3f4847a3ae1011f16065fa64651356f9
  Building wheel for numpy (pyproject.toml) ... done
  Created wheel for numpy: filename=numpy-1.19.1-cp39-cp39-linux_x86_64.whl size=13360454 sha256=6c0b4703366a549b7f7bdd379178aa18d8e5227bc8fa1ed6ae539c79feab813d
  Stored in directory: /root/.cache/pip/wheels/e0/85/4b/385d1720e1dd8bee99bac183aac3454e88aba9fc3146ccfc27
Successfully built pyroomacoustics scipy numpy
Failed to build pandas
ERROR: Could not build wheels for pandas, which is required to install pyproject.toml-based projects`
I am running in colab.

**Reply by mpariente** [reply]
Try installing before. Failure to install pandas has nothing to do with asteroid.

**Comment by xs20211015** [comment, marked as answer]
In addition, I also tried to run two other models on colab. But have encountered problems.
First:
```
%cd /content/asteroid/egs/wham/ConvTasNet
!./run.sh
```
I set the path of storage_dir in run.sh and start from stage 0.
But when run to here: $python_path create_wham_from_scratch.py
I first solved this error:
```
usage: create_wham_from_scratch.py
        [-h]
        [--output-dir OUTPUT_DIR]
        [--wsj0-root WSJ0_ROOT]
        [--wham-noise-root WHAM_NOISE_ROOT]
create_wham_from_scratch.py: error: argument --wsj0-root: expected one argument
```
I found that the path to $wav_dir was not recognized, I solved it by putting wav_dir=tmp after mkdir -p $out_dir.
Then after running into create_wham_from_scratch.py, this error was reported:
```
Traceback (most recent call last):
   File "/content/asteroid/egs/wham/ConvTasNet/tmp/wham_scripts/create_wham_from_scratch.py", line 118, in <module>
     create_wham(args.wsj0_root, args.wham_noise_root, args.output_dir)
   File "/content/asteroid/egs/wham/ConvTasNet/tmp/wham_scripts/create_wham_from_scratch.py", line 34, in create_wham
     scaling_npz = np.load(scaling_npz_path, allow_pickle=True)
   File "/usr/local/lib/python3.9/dist-packages/numpy/lib/npyio.py", line 407, in load
     fid = stack. enter_context(open(os_fspath(file), "rb"))
FileNotFoundError: [Errno 2] No such file or directory: 'tmp/wham_noise/metadata/scaling_tr.npz'
```
I found out it was a file path issue. I changed the paths of --wsj0-root and other parameters to absolute paths to solve this error.
Next, when still running create_wham_from_scratch.py, an error is reported at the line s1 = read_scaled_wav(s1_path, scaling_wsjmix[i_utt][0], downsample):
```
Traceback (most recent call last):
   File "/content/asteroid/egs/wham/ConvTasNet/tmp/wham_scripts/create_wham_from_scratch.py", line 118, in <module>
     create_wham(args.wsj0_root, args.wham_noise_root, args.output_dir)
   File "/content/asteroid/egs/wham/ConvTasNet/tmp/wham_scripts/create_wham_from_scratch.py", line 79, in create_wham
     s1 = read_scaled_wav(s1_path, scaling_wsjmix[i_utt][0], downsample)
   File "/content/asteroid/egs/wham/ConvTasNet/tmp/wham_scripts/utils.py", line 7, in read_scaled_wav
     samples, sr_orig = sf. read(path)
   File "/usr/local/lib/python3.9/dist-packages/soundfile.py", line 285, in read
     with SoundFile(file, 'r', samplerate, channels,
   File "/usr/local/lib/python3.9/dist-packages/soundfile.py", line 658, in __init__
     self._file = self._open(file, mode_int, closefd)
   File "/usr/local/lib/python3.9/dist-packages/soundfile.py", line 1216, in _open
     raise LibsndfileError(err, prefix="Error opening {0!r}: ".format(self.name))
soundfile.LibsndfileError: Error opening '/content/asteroid/egs/wham/ConvTasNet/tmp/wsj0/si_tr_s/40n/40na010x.wav': System error.
```
I checked my path /content/asteroid/egs/wham/ConvTasNet/tmp and wsj0/si_tr_s/40n/40na010x.wav does not exist. Is this caused by the fact that I don't have the wsj0 dataset? I tried a few things, but couldn't solve the problem.

**Reply by mpariente** [reply]
> Is this caused by the fact that I don't have the wsj0 dataset ? 

Yes, this is it.

**Reply by xs20211015** [reply]
Ok, thanks again for your reply.
The models I tried: ConvTasNet, DPRNN and TasNet should all be related to the wsj0 dataset. I don't have this dataset at the moment, I may have to try the model with other datasets. Thank you for your patient answer!
